plugins/gfm/src/main/kotlin/org/jetbrains/dokka/gfm/renderer/CommonmarkRenderer.kt

```diff
@@ -35,6 +36,11 @@ open class CommonmarkRenderer(
                 childrenCallback()
                 buildParagraph()
             }
+            node.dci.kind == ContentKind.BriefComment -> {
```

Comment:
We shouldn't switch by the kind here. I know it's done this way in several places, but it's not a good solution. If the brief comment requires this sequence, it should have a `TextStyle.Block` or some similar one added, possibly by a format-specific transformer

Reply:
Transformer - too complex for me, sorry :) 
Maximum - can add a comment with TODO.

And what is wrong with that, in HtmlRenderer it's wrapped in `div`, and `div` is block-level element anyway
```kotlin
node.dci.kind == ContentKind.BriefComment -> div("brief $additionalClasses") { childrenCallback() }
```

It's needed there because without it properties table 
> | Name | Summary |
> |---|---|
> | [padded](padded.md) | `var padded: Boolean`<br>If `true`, the container insert some space between children. |
> | [stretchy](stretchy.md) | `var stretchy: Boolean = false`<br>Next added child should expand to use all available size. |
```
| [padded](padded.md) | `var padded: Boolean`<br>If `true`, the container insert some space between children. |
```

renders without \<br> and even spaces:
> | Name | Summary |
> |---|---|
> | [padded](padded.md) | `var padded: Boolean`If `true`, the container insert some space between children. |
> | [stretchy](stretchy.md) | `var stretchy: Boolean = false`Next added child should expand to use all available size. |
```
| [padded](padded.md) | `var padded: Boolean`If `true`, the container insert some space between children. |
```

Reply:
It's unmaintainable, as HtmlRenderer shows, and wasn't meant to be used there. Kinds should never be used at this point, they should only be used in transformers for identifying the actual element (eg. finding all signatures in the ContentNodes tree). That's my fault that they are accessible here though. 
I'll fix this part then, thanks!

Reply:
Thanks @kamildoleglo,  9895b2e works as expected.
So this is resolved?

Reply:
Yup, it is. Let me get some performance measurements and we'll probably get that merged